docker/mining.Dockerfile

```diff
@@ -0,0 +1,196 @@
+FROM chef AS planner
+COPY . .
+RUN cargo chef prepare --recipe-path recipe.json
+
+# In this stage we download all system requirements to build the project
+#
+# It also captures all the build arguments to be used as environment variables.
+# We set defaults for the arguments, in case the build does not include this information.
+FROM chef AS deps
+SHELL ["/bin/bash", "-xo", "pipefail", "-c"]
+COPY --from=planner /opt/zebrad/recipe.json recipe.json
+
+# Install zebra build deps
+RUN apt-get -qq update && \
+    apt-get -qq install -y --no-install-recommends \
+    llvm \
+    libclang-dev \
+    clang \
+    ca-certificates \
+    protobuf-compiler \
+    ; \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+# Install google OS Config agent to be able to get information from the VMs being deployed
+# into GCP for integration testing purposes, and as Mainnet nodes
+# TODO: this shouldn't be a hardcoded requirement for everyone
+RUN if [ "$(uname -m)" != "aarch64" ]; then \
+      apt-get -qq update && \
+      apt-get -qq install -y --no-install-recommends \
+      curl \
+      lsb-release \
+      && \
+      echo "deb http://packages.cloud.google.com/apt google-compute-engine-$(lsb_release -cs)-stable main" > /etc/apt/sources.list.d/google-compute-engine.list && \
+      curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add - && \
+      apt-get -qq update  && \
+      apt-get -qq install -y --no-install-recommends google-osconfig-agent; \
+    fi \
+    && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+# Build arguments and variables set to change how tests are run, tracelog levels,
+# and Network to be used (Mainnet or Testnet)
+#
+# We set defaults to all variables.
+ARG RUST_BACKTRACE
+ENV RUST_BACKTRACE ${RUST_BACKTRACE:-0}
+
+ARG RUST_LIB_BACKTRACE
+ENV RUST_LIB_BACKTRACE ${RUST_LIB_BACKTRACE:-0}
+
+ARG COLORBT_SHOW_HIDDEN
+ENV COLORBT_SHOW_HIDDEN ${COLORBT_SHOW_HIDDEN:-0}
+
+ARG RUST_LOG
+ENV RUST_LOG ${RUST_LOG:-info}
+
+# Skip IPv6 tests by default, as some CI environment don't have IPv6 available
+ARG ZEBRA_SKIP_IPV6_TESTS
+ENV ZEBRA_SKIP_IPV6_TESTS ${ZEBRA_SKIP_IPV6_TESTS:-1}
+
+# Use default checkpoint sync and network values if none is provided
+ARG CHECKPOINT_SYNC
+ENV CHECKPOINT_SYNC ${CHECKPOINT_SYNC:-true}
+
+ARG NETWORK
+ENV NETWORK ${NETWORK:-Testnet}
+
+ENV CARGO_HOME /opt/zebrad/.cargo/
+
+# In this stage we build tests (without running then)
+#
+# We also download needed dependencies for tests to work, from other images.
+# An entrypoint.sh is only available in this step for easier test handling with variables.
+FROM deps AS tests
+# TODO: do not hardcode the user /root/ even though is a safe assumption
+# Pre-download Zcash Sprout, Sapling parameters and Lightwalletd binary
+COPY --from=us-docker.pkg.dev/zealous-zebra/zebra/zcash-params /root/.zcash-params /root/.zcash-params
+COPY --from=us-docker.pkg.dev/zealous-zebra/zebra/lightwalletd /opt/lightwalletd /usr/local/bin
```

Comment:
`lightwalletd` is not required for mining, please make it conditional on "not mining" in the original Dockerfile.

Reply:
I had some trouble handling booleans so I added `FEATURES` and `TEST_FEATURES` ARGS in [`63158a4`](https://github.com/ZcashFoundation/zebra/pull/6228/commits/63158a4b1daa80b7c437afc4c0d89dfbad1a542b) for configuring features. `TEST_FEATURES` is 'lightwalletd-grpc-tests` by default, so it's built with FEATURES="getblocktemplate-rpcs" and TEST_FEATURES="" in [the docker build workflow](https://github.com/ZcashFoundation/zebra/pull/6228/commits/63158a4b1daa80b7c437afc4c0d89dfbad1a542b#diff-5b40193e09025fd041de92d38c976d0b469e3ee85bef45a71d05d8b5078b1ee8R164) to skip the tests.

Reply:
Thanks, that seems like a good solution!